py-polars/tests/unit/io/test_delta.py

```diff
@@ -7,6 +7,7 @@
 import pyarrow.fs
 import pytest
 from deltalake import DeltaTable
+from deltalake._internal import TableNotFoundError
```

Comment:
It's not great that we have to import this from a private module. I would recommend that deltalake exposes their exceptions in a public module, to allow users to catch and handle such exceptions.

Doesn't have to block this PR though, just an observation.

Reply:
Apparently it was already under the exceptions module, I just missed it :)